applications/ALEapplication/test_examples/01_Mesh_Test_2D3N.gid/MainKratos.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+from math import sin,cos
+
+#Definition of functions
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+movementList = ["TRANSLATION_X", "TRANSLATION_Y", "BENDING", "ROTATION"]
+# =============================================================
+movement = 0             # set 0-3
+amplificationFactor = 2  # factor to amplify the movement
+solveMesh = True         # set to "True" to solve the mesh
+# =============================================================
+
+
+#Apply displacement boundary conditions
+
+def ApplyDisplacementConditions(main_model_part):
+    for node in main_model_part.GetSubModelPart("Inlet2D_Inlet").Nodes: 
+        node.Fix(MESH_DISPLACEMENT_X)
+        node.Fix(MESH_DISPLACEMENT_Y)
+    for node in main_model_part.GetSubModelPart("Outlet2D_Outlet").Nodes: 
+        node.Fix(MESH_DISPLACEMENT_Y)
+        node.Fix(MESH_DISPLACEMENT_X)
+    for node in main_model_part.GetSubModelPart("Slip2D_Walls").Nodes:  
+        node.Fix(MESH_DISPLACEMENT_Y)
+        node.Fix(MESH_DISPLACEMENT_X)
+    for node in main_model_part.GetSubModelPart("NoSlip2D_Interface").Nodes:
+        node.Fix(MESH_DISPLACEMENT_X)
+        node.Fix(MESH_DISPLACEMENT_Y)
+      
+def DisplacementToMesh(fluid_interface, time, movement, ampFac):
+    time *= ampFac
+    for node in fluid_interface:
+        if movement == "TRANSLATION_X":
+            print("Setting Translation")
+            valueX = 0.5 * time
+            valueY = 0
+        elif movement == "TRANSLATION_Y":
+            valueX = 0
+            valueY = 0.2 * time
+        elif movement == "BENDING":
+            valueX = 0
+            valueY = 3 * node.X0 * node.X0 * time
+        elif movement == "ROTATION":
+            xOld = node.X0
+            yOld = node.Y0
+            valueX = (cos(time*.6)*xOld + sin(time*.6)*yOld) - xOld
+            valueY = (-sin(time*.6)*xOld + cos(time*.6)*yOld) - yOld
+
+        else:
+            wait = input("Wrong type of movement specified, please correct input")
+        # set the prescribed values
+        node.SetSolutionStepValue(MESH_DISPLACEMENT_X,0,valueX)
+        node.SetSolutionStepValue(MESH_DISPLACEMENT_Y,0,valueY)
+      
+
+#Set number of threads
+
+def SetParallelSize(num_threads):
```

Comment:
please do not use this function. number of threads should be set by employing env vars

export OMP_NUM_THREADS=2 (to have 2 threads)

this has been discussed at length and the rationale is that otherwise you never know if someone is setting the number of threads somewhere in the code